rideshare.rb

```diff
@@ -0,0 +1,136 @@
+rides = [
+  {
+    driver_id: "DR0004",
+    date: "3rd Feb 2016",
+    cost: 5,
+    rider_id: "RD0022",
+    rating: 5,
+  },
+  {
+    driver_id: "DR0001",
+    date: "3rd Feb 2016",
+    cost: 10,
+    rider_id: "RD0003",
+    rating: 3,
+  },
+  {
+    driver_id: "DR0002",
+    date: "3rd Feb 2016",
+    cost: 25,
+    rider_id: "RD0073",
+    rating: 5,
+  },
+  {
+    driver_id: "DR0001",
+    date: "3rd Feb 2016",
+    cost: 30,
+    rider_id: "RD0015",
+    rating: 4,
+  },
+  {
+    driver_id: "DR0003",
+    date: "4th Feb 2016",
+    cost: 5,
+    rider_id: "RD0066",
+    rating: 5,
+  },
+  {
+    driver_id: "DR0004",
+    date: "4th Feb 2016",
+    cost: 10,
+    rider_id: "RD0022",
+    rating: 4,
+  },
+  {
+    driver_id: "DR0002",
+    date: "4th Feb 2016",
+    cost: 15,
+    rider_id: "RD0013",
+    rating: 1,
+  },
+  {
+    driver_id: "DR0003",
+    date: "5th Feb 2016",
+    cost: 50,
+    rider_id: "RD0003",
+    rating: 2,
+  },
+  {
+    driver_id: "DR0002",
+    date: "5th Feb 2016",
+    cost: 35,
+    rider_id: "RD0066",
+    rating: 3,
+  },
+  {
+    driver_id: "DR0004",
+    date: "5th Feb 2016",
+    cost: 20,
+    rider_id: "RD0073",
+    rating: 5,
+  },
+  {
+    driver_id: "DR0001",
+    date: "5th Feb 2016",
+    cost: 45,
+    rider_id: "RD0003",
+    rating: 2,
+  },
+]
+
+# find the number of rides each driver has given
+# using @ allows me to shorthand use driver_rides in the other methods
+def number_of_rides(rides_array, driver_id)
+  @driver_rides = rides_array.select do |ride|
```

Comment:
From now on, we'll try to avoid using `@` unless we're intentionally making instance variables in classes!